### PR TITLE
Migrate wallet

### PR DESCRIPTION
**Pull request by katzio**

### CODE REVIEW

**Comment by katzio** [comment]
We might not want to merge duplicated code with wallet creation at leave it as is...merging will make a bit of a messy code

File: app/src/main/res/layout/boot_wallet_error_fragment.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <data xmlns:android="http://schemas.android.com/apk/res/android">
         <variable
             name="model"
-            type="org.kinecosystem.kinit.viewmodel.CreateWalletViewModel"/>
+            type="org.kinecosystem.kinit.view.bootWallet.BootWalletActions"/>
```
Comment (berryve) [inline app/src/main/res/layout/boot_wallet_error_fragment.xml]: BootWalletActions is the model ?? @katzio

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -68,7 +68,7 @@
             android:windowSoftInputMode="adjustPan" />
 
         <activity
-            android:name=".view.createWallet.CreateWalletActivity"
+            android:name=".view.bootWallet.BootWalletActivity"
```
Comment (berryve) [inline app/src/main/AndroidManifest.xml]: naming convention for package names is all lower case.
i.e. bootwallet

File: app/src/main/res/layout/boot_wallet_error_fragment.xml (hunk 1 above)
Comment (berryve) [reply app/src/main/res/layout/boot_wallet_error_fragment.xml]: suggest to rename to BootWalletViewModel

File: app/src/main/res/layout/migrate_wallet_complete_fragment.xml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (berryve) [inline app/src/main/res/layout/migrate_wallet_complete_fragment.xml]: could this not also be merged with create_wallet_complete_fragment.xml ?

File: app/src/main/res/layout/migrate_wallet_error_fragment.xml [hunk 4]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (berryve) [inline app/src/main/res/layout/migrate_wallet_error_fragment.xml]: could this not be merged with create_wallet_error_fragment.xml ?

File: app/src/main/res/layout/boot_wallet_error_fragment.xml (hunk 1 above)
Comment (katzio) [reply app/src/main/res/layout/boot_wallet_error_fragment.xml]: that's no a model, that's an interface of 2 models - CreateWalletViewModel & MigrateWalletViewModel

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (katzio) [reply app/src/main/AndroidManifest.xml]: 👍

File: app/src/main/res/layout/migrate_wallet_complete_fragment.xml (hunk 3 above)
Comment (katzio) [reply app/src/main/res/layout/migrate_wallet_complete_fragment.xml]: 👍

File: app/src/main/res/layout/migrate_wallet_error_fragment.xml (hunk 4 above)
Comment (katzio) [reply app/src/main/res/layout/migrate_wallet_error_fragment.xml]: 👍

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -66,7 +67,10 @@
     <string name="kin_currency">KIN</string>
     <string name="version">version %s</string>
     <string name="error_unable_to_create_wallet">We were unable to create a wallet for you</string>
-    <string name="error_create_wallet">Please restart the app to try again.</string>
+    <string name="error_unable_to_migrate_wallet">We were unable to migrate a wallet for you</string>
```
Comment (berryve) [inline app/src/main/res/values/strings.xml]: do we need this string? can we remove it ?

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -66,7 +67,10 @@
...
+
+    <string name="error_unable_to_boot_wallet_title">Sorry, error occurred. </string>
```
Comment (berryve) [inline app/src/main/res/values/strings.xml]: where is this used?

File: app/src/main/java/org/kinecosystem/kinit/blockchain/Wallet.kt [hunk 7]
```diff
@@ -408,4 +443,28 @@ class Wallet(context: Context, dataStoreProvider: DataStoreProvider,
         categoriesRepository.updateCurrentTaskState(taskId, state)
     }
 
+    fun migrateWallet(migrationManagerCallbacks: IMigrationManagerCallbacks? = null) {
+        migrationManager?.start(object : IMigrationManagerCallbacks {
+            override fun onReady(newKinClient: IKinClient) {
+                Log.d("WalletMigration", "migrateWallet onReady for account: ${userRepo.userId()} " +
+                        "with public address ${kinClient.getAccount(0)?.publicAddress}")
+                isKin3 = true
```
Comment (berryve) [inline app/src/main/java/org/kinecosystem/kinit/blockchain/Wallet.kt]: should set this to true also when new wallet is created

**Pull request merged**